stl/inc/charconv

```diff
@@ -448,7 +448,9 @@ _NODISCARD inline uint32_t _Bit_scan_reverse(const _Big_integer_flt& _Xval) noex
 
     unsigned long _Index; // Intentionally uninitialized for better codegen
 
-    _BitScanReverse(&_Index, _Xval._Mydata[_Bx]); // assumes _Xval._Mydata[_Bx] != 0
+    // assumes _Xval._Mydata[_Bx] != 0
+    _BitScanReverse(
+        &_Index, _Xval._Mydata[_Bx]); // lgtm [cpp/conditionallyuninitializedvariable] _Xval._Mydata[_Bx] != 0

```

Comment:
If using `__assume` works, I would rather do this. 
Or ditto if the following works:
```C++
auto blah = _Xval._Mydata[_Bx];
if (blah != 0) {
    _BitScanReverse(
        &_Index, _Xval._Mydata[_Bx]);
} else {
    _Unreachable();
}
```
Sure by _works_ I imply not making codegen worse and suppressing `codeql` warnings.